site/pages/retirements/[beneficiary_address]/index.tsx

```diff
@@ -31,12 +37,23 @@ export const getStaticProps: GetStaticProps<PageProps, Params> = async (
       throw new Error("No params found");
     }
 
+    let addressByDomain;
+    const isDomainInURL = getIsDomainInURL(params.beneficiary_address);
+    if (isDomainInURL) {
+      addressByDomain = await getAddressByDomain(params.beneficiary_address);
+    }
+
+    // Do not continue if KNS or ENS Domain can not be resolved
+    if (addressByDomain === null) {
```

Comment:
makes me uncomfortable that `undefined` is OK (not a domain) but `null` is an error.

What about something like this up on L40?
```ts
let resolvedAddress: string;
if (isDomainInURL(params.beneficiary_address) {
  resolvedAddress = await getAddressByDomain(params.beneficiary_address); // this fn should throw if it fails to resolve
} else if (ethers.utils.isAddress(address)) {
  resolvedAddress = params.beneficiary_address;
} else {
  throw new Error("Not a valid beneficiary address");
}
```

Reply:
Thanks for the hint with the ether helper. Used in multiple places now. Done